spec/models/user_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validation test', :aggregate_failures do
+    it 'username presence' do
+      user = User.new(password: 'password', password_confirmation: 'password').save
+      expect(user).to eq(false)
```

Comment:
let add a new line to separate the `setup phase` and the `expectating phase` @gutakk 

```
user = User.new(password: 'password', password_confirmation: 'password').save

expect(user).to eq(false)
```

Another thing, instead of using `User.new(password: 'password', password_confirmation: 'password').save` we could use `User.create(password: 'password', password_confirmation: 'password')`

Reply:
I used `new` + `save` because I want to expect if user can save to db or not.
- `save` return `true` or `false`
- `create` return model instance
If I use `create` I have to check something like `user1.valid?`
What do you think?

Reply:
IMO, `user == true` looks weird 🤔 the user variable mean the user object, not a boolean object 🤔

if you would like to check the true/flase, I think we can go with

```
expect(User.new(password: 'password', password_confirmation: 'password').save). to eq(false)
```


or there is another way

```
user = User.create(xxx)

expect(user).to be_truly
```

For more information, you can check on https://github.com/rspec/rspec-expectations#truthiness

Reply:
Make sense now. I will make the change.

Reply:
Well this spec was updated to use `shoulda-matcher`.

Reply:
👍